### ISSUE TITLE
Netty server span name doesn't contain the http route

### ISSUE THREAD

**Issue by Samudraneel24** [open-telemetry/opentelemetry-java-instrumentation#6050]
The Netty server span name is "HTTP method", unlike some other instrumentations such as Servlet where the span name contains the http route which seems to be more helpful for users. Can we have similar span names for Netty instrumentation as well?

**Comment by laurit** [comment]
Nobody has yet figured out how to extract a route from netty request. The chances of extracting a route are much better with higher level frameworks that are often built on netty. If you are using such a framework I'd consider checking whether it is possible to get a route there and if so implementing an instrumentation for that framework.

**Comment by Samudraneel24** [comment]
In Netty, in the `HttpRequest` class we have a `getUri()` method. So we have access to the URI. We could then parse it to remove the parameters and get just the http path, I'll try to find a way to do this. We can set the path as the span name. Do you see any potential issues with this approach?

**Comment by laurit** [comment]
Unfortunately [opentelemetry specification](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md#name) states that
```
Instrumentation MUST NOT default to using URI path as span name
```

**Comment by Samudraneel24** [comment]
Does this mean that we can't use the exact URI as the span name, or that we can't use the URI to derive the span name from?

**Comment by laurit** [comment]
> Does this mean that we can't use the exact URI as the span name, or that we can't use the URI to derive the span name from?

You are correct, according to the spec URI should not be used as a span name. Span names have to have low cardinality, URI could be anything. It could be something like `get_product/1234` where product id is part of the URI, or somebody could be requesting for stuff that does not exist.
Deriving span name from URI is allowed. For example `get_product/{product_id}` would be a good span name. The problem here is how do you derive such a span name?

**Comment by Samudraneel24** [comment]
Got it. You're correct, I hadn't considered a case like `get_product/1234`.
I'll try to find if there's any work around this.

**Comment by Samudraneel24** [comment]
I haven't found any exact solution for this, although there can be some approximate solutions to remove the high cardinality portion of the URL.
For instance only keep the segments starting from left to right, which are alphabetic, i.e. don't contain any numerals and remove all the following segments. If there are no such alphabetic segments in the URL then we can have the span name as "HTTP method".
So, is a solution like this something we want or should we stick to the current span naming?

**Comment by trask** [comment]
Alphabetic segments can also be problematic, e.g. usernames, so I'm not sure we would want to do this by default, but having some kind of configurable (regex-based?) mapping from paths to span names could be nice.

**Comment by Samudraneel24** [comment]
We can have a system property through which we provide the number of URI segments to be used for naming the span. This gives the control of span naming to the user. The default behaviour can be the current naming, i.e. "HTTP method", so only if the user provides configuration we use the new span naming.

**Comment by Samudraneel24** [comment]
Any thoughts on this?

**Comment by trask** [comment]
hi @Samudraneel24, if you are using an underlying controller that the javaagent instruments, I believe it should be updating the span name and `http.route` now in recent javaagent versions (though these are populated too late to be useful in sampling), can you confirm your underlying scenario?
